File: src/main/java/com/parth/quizapp/Model/Quiz.java
package com.parth.quizapp.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Quiz {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String description;

    private int maxMarks;

    private int numberOfQuestions;

    private boolean active = false;

    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "creator_id")
    private User creator;

    @OneToMany(mappedBy = "quiz", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Ques> questions = new ArrayList<>();

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
    }
}

File: quiz-frontend/src/types/quiz.ts
export interface QuizDTO {
  id?: number;
  code?: string;
  title: string;
  description: string;
  maxMarks: number;
  numberOfQuestions: number;
  active?: boolean;
}

export interface QuizFormData {
  title: string;
  code?: string;
  description: string;
  maxMarks: number;
  numberOfQuestions: number;
  active: boolean;
}
